paywall/src/utils/durations.js

```diff
@@ -62,13 +62,15 @@ export function durationsAsTextFromSeconds(seconds) {
 export function durationsAsArrayFromSeconds(seconds) {
   const d = durations(seconds, {})
   const asArrayOfValues = Object.keys(d).map(duration => {
-    if (d[duration] !== 1) {
+    const durationFloor = Math.floor(d[duration])
+    if (durationFloor === 0) return ''
+    if (durationFloor !== 1) {
       // Singular should only be used when there is exactly 1; otherwise plural is needed
-      return `${d[duration]} ${duration}`
+      return `${durationFloor} ${duration}`
     }
-    return `${d[duration]} ${duration.slice(0, -1)}` // remove the s!
+    return `${durationFloor} ${duration.slice(0, -1)}` // remove the s!
   })
-  return asArrayOfValues
+  return asArrayOfValues.filter(a => a) // remove empty entries
```

Comment:
Should we encounter empty entries here?  Can we update the caller to filter empties prior to passing to this method to reduce complexity here?

Reply:
Nope. see line 66.

Reply:
I am not sure that I understand, would you be able to elaborate a bit more?

Reply:
sure. This is to prevent "1 hour and 0 seconds" which will happen if we have (for example) "1 hour and 0.23432423 seconds" before flooring. The solution I chose is to simply map 0 to an empty string, and remove them. If we don't remove them, it becomes "1 hour and "